### DISCUSSION TITLE
`CS8785: Generator 'MapperGenerator' failed to generate source` buliding Mapperly

### DISCUSSION THREAD

**Discussion by peteraritchie** [riok/mapperly#473] [Q&A] [answered]
When I clone the repo, open Visual Studio, load the solution, and build I get this error in the Sample and IntegrationTests projects.  It hasn't stopped me from contributing, and I figured it was a circular dependency kind of thing.  But I'm starting to think it's an indication of a problem.  Am I over thinking that?

**Comment by latonz** [comment]
Thank you for contributing despite the difficulties! I don't have access to Visual Studio as I don't use Windwos, @CommonGuy or @TimothyMakkison can you reproduce this?

**Reply by TimothyMakkison** [reply]
Yeah had this for a while [see here.](https://github.com/riok/mapperly/pull/419#issuecomment-1560181834) I think I could still build older branches but never pinpointed the cause. Could be a VS or .NET issue, note that rider doesn't have this issue. 

`[Warning] Found project reference without a matching metadata reference: C:\Users\timma\Desktop\coding\C#\mapperly\src\Riok.Mapperly.Abstractions\Riok.Mapperly.Abstractions.csproj`

`Generator 'MapperGenerator' failed to generate source. Exception was of type 'InvalidOperationException' with message 'Type Riok.Mapperly.Abstractions.EnumMappingStrategy, Riok.Mapperly.Abstractions not found`

**Comment by CommonGuy** [comment, marked as answer]
I do not have Visual Studio either. What version of VS are you using? Is there more information in the error message? Does building the solution via `dotnet build` work?

It seems that Visual Studio still struggles with source generator support from what I've read. Some users report that closing VS, deleting all `bin` and `obj` folders, then restarting VS solves the problem.

**Reply by TimothyMakkison** [reply]
> Does building the solution via dotnet build work?

`dotnet run` definitely still works

update: I can use `dotnet test` and run the sample with `dotnet run` on the main branch. I still can't use VS.

**Reply by peteraritchie** [reply]
Sometimes VS seems to grok that the errors are from a generator and hides them in the error list.  I'm using VS 17.6.2.  `dotnet build` and `dotnet test` work fine.  I'll chalk it up to a VS _idiosyncrasy_.
